tests/unit/Rules/AbstractCompositeTest.php

```diff
@@ -150,4 +150,77 @@ public function testShouldNotDefineNameForInternalRulesWhenItHasAName()
         $compositeRuleMock->addRule($simpleRuleMock);
         $compositeRuleMock->setName($ruleName);
     }
+
+    public function testRemoveRulesShouldRemoveAllTheAddedRules()
+    {
+        $simpleRuleMock = $this->getMock('Respect\\Validation\\Validatable');
+
+        $compositeRuleMock = $this->getMockForAbstractClass('Respect\\Validation\\Rules\\AbstractComposite');
+        $compositeRuleMock->addRule($simpleRuleMock);
+        $compositeRuleMock->removeRules();
+
+        $this->assertEmpty($compositeRuleMock->getRules());
+    }
+
+    public function testShouldReturnTheAmountOfAddedRules()
+    {
+        $compositeRuleMock = $this->getMockForAbstractClass('Respect\\Validation\\Rules\\AbstractComposite');
+        $compositeRuleMock->addRule($this->getMock('Respect\\Validation\\Validatable'));
+        $compositeRuleMock->addRule($this->getMock('Respect\\Validation\\Validatable'));
+        $compositeRuleMock->addRule($this->getMock('Respect\\Validation\\Validatable'));
+
+        $this->assertCount(3, $compositeRuleMock->getRules());
+    }
+
+    public function testHasRuleShouldReturnFalseWhenThereIsNoRuleAppended()
+    {
+        $compositeRuleMock = $this->getMockForAbstractClass('Respect\\Validation\\Rules\\AbstractComposite');
+
+        $this->assertFalse($compositeRuleMock->hasRule(''));
+    }
+
+    public function testHasRuleShouldReturnFalseWhenRuleIsNotFound()
+    {
+        $oneSimpleRuleMock = $this->getMock('Respect\\Validation\\Validatable');
+
+        $compositeRuleMock = $this->getMockForAbstractClass('Respect\\Validation\\Rules\\AbstractComposite');
+        $compositeRuleMock->addRule($oneSimpleRuleMock);
+
+        $anotherSimpleRuleMock = $this->getMock('Respect\\Validation\\Validatable');
+
+        $this->assertFalse($compositeRuleMock->hasRule($anotherSimpleRuleMock));
+    }
+
+    public function testHasRuleShouldReturnFalseWhenRulePassedAsStringIsNotFound()
+    {
+        $simpleRuleMock = $this->getMock('Respect\\Validation\\Validatable');
+
+        $compositeRuleMock = $this->getMockForAbstractClass('Respect\\Validation\\Rules\\AbstractComposite');
+        $compositeRuleMock->addRule($simpleRuleMock);
+
+        $this->assertFalse($compositeRuleMock->hasRule('SomeRule'));
+    }
+
+    public function testHasRuleShouldReturnTrueWhenRuleIsFound()
+    {
+        $simpleRuleMock = $this->getMock('Respect\\Validation\\Validatable');
+
+        $compositeRuleMock = $this->getMockForAbstractClass('Respect\\Validation\\Rules\\AbstractComposite');
+        $compositeRuleMock->addRule($simpleRuleMock);
+
+        $this->assertTrue($compositeRuleMock->hasRule($simpleRuleMock));
```

Comment:
I would do here another *assertion* trying to check if there's a rule using a `string`. but I couldn't be able to do that because phpunit mocks are not working so fine with `get_class()` which is used internally in the method (https://github.com/Respect/Validation/blob/master/library/Rules/AbstractComposite.php#L92).

btw it's a known issue on phpunit repo:  https://github.com/sebastianbergmann/phpunit-mock-objects/issues/134